.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+name: Tests
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    name: Build snap
+    uses: canonical/data-platform-workflows/.github/workflows/build_snap.yaml@v23.0.4
+  
+  test:
+    name: Test snap
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Download snap package(s)
+        uses: actions/download-artifact@v4
+        with:
+          pattern: ${{ needs.build.outputs.artifact-prefix }}-*
+          merge-multiple: true
+      
+      - name: Install snap package(s)
+        run: |
+          sudo snap install --dangerous charmed-etcd*.snap
+      
+      - name: Check etcd service status and run etcdctl and etcdutl
+        run: |
+          sudo systemctl status snap.charmed-etcd.daemon.service | grep 'active (running)'
+          charmed-etcd.etcd --version
+          charmed-etcd.etcdctl version
+          charmed-etcd.etcdutl version
+      
+      - name: Run tests
+        run: |
+          charmed-etcd.etcdctl endpoint health | grep 'is healthy'
+          charmed-etcd.etcdctl put greeting "Hello, etcd" | grep 'OK'
+          charmed-etcd.etcdctl get greeting | grep 'Hello, etcd'
+      
+      - name: Run Performance checks
+        run: |
+          charmed-etcd.etcdctl check perf
```

Comment:
What is the output / outcome of this check?

Reply:
This runs a series of write/read and returns the throughput and the slowest request time.